Documentation/Architecture/SpatialAwarenessSystemArchitecture.md

```diff
@@ -0,0 +1,621 @@
+# Mixed Reality Toolkit Spatial Awareness System Architecture
+
+The Mixed Reality Spatial Awareness System is intended to abstract the specific implementation details of platforms, such as the Microsoft HoloLens, that provide support for mapping the real-world and providing the data to Mixed Reality experiences.
+
+As of this writing, the Mixed Reality Toolkit will provide support for Spatial Awareness on the Microsoft HoloLens. As additional platforms are supported, such as ARKit and ARCore, this support will be expanded to encompass any features and/or limitations of those systems.
+
+Where possible, we are designing for common functionality (ex: identification of planar surfaces) well as access to the lowest level of available data (ex: mesh or point data).
+
+It is expected that many platforms may not support one or more of the interfaces defined herein. In fact, some may not support spatial awareness at all. On those platforms, the system must gracefully fail and provide the developer with appropriate data (null, empty collections, etc.) in return.
+
+Each interface defined will implement one or more Properties, Methods and/or Events (PMEs) that can be accessed by application code.
+
+<img src="Images/SpatialAwarenessSystemArchitecture.png">
+
+## Interfaces
+
+## IMixedRealitySpatialAwarenessSystem
+
+| Toolkit Layer | Namespace |
+| --- | --- |
+| Core | Microsoft.MixedReality.Toolkit.Internal.Interfaces.SpatialAwarenessSystem |
```

Comment:
Do we want to update this to be `Internal` to reflect the planned namespace changes?

Reply:
For this PR, I am sticking with the current namespace truth.